roles/vault/tasks/vault.yml

```diff
@@ -30,10 +30,10 @@
   run_once: True
   no_log: True
   when: not vault_init_status.json.initialized
-  register: vault_init_output
+  register: vault_keys
```

Comment:
What format does this take? Seems like it's going to be different than the YAML output we had before.

Reply:
Previously:

https://github.com/stackhpc/ansible-collection-hashicorp/runs/6833603104?check_suite_focus=true#step:5:41

2022-06-10T15:24:01.0709228Z TASK [vault : Print vault keys] ************************************************
2022-06-10T15:24:01.0984811Z ok: [localhost] => {
2022-06-10T15:24:01.0985412Z     "vault_keys": {
2022-06-10T15:24:01.0987019Z         "recovery_keys_b64": [],
2022-06-10T15:24:01.0987415Z         "recovery_keys_hex": [],
2022-06-10T15:24:01.0987812Z         "recovery_keys_shares": 5,
2022-06-10T15:24:01.0988224Z         "recovery_keys_threshold": 3,
2022-06-10T15:24:01.0988723Z         "root_token": "hvs.aGoMsJWinxfIRM8bh0eK8r8g",
2022-06-10T15:24:01.0990639Z         "unseal_keys_b64": [
2022-06-10T15:24:01.0991150Z             "Cj3QdQM/Fl/Hs6WZEmbdbGhmuuJwg7MgxPvSBHGCyALp",
2022-06-10T15:24:01.0991640Z             "cM7S8Wwy3MhelwHUwFSX4yv+XMIjC3G6XuVUSivlLGpG",
2022-06-10T15:24:01.1040793Z             "ZLy9vjXlAK9fRWiXYEItdfDZipX+QevIKyI3GJhFQIAH",
2022-06-10T15:24:01.1041176Z             "CXrVtbaAwbMxj/2Cw8RnN1IG1QJRRgs6quX10UFzTmiA",
2022-06-10T15:24:01.1041558Z             "e5ExvosieoRpLPIGcRr8S4lyaGeyrFhGZ4TbHx9E10M1"
2022-06-10T15:24:01.1041845Z         ],
2022-06-10T15:24:01.1042041Z         "unseal_keys_hex": [
2022-06-10T15:24:01.1042356Z             "0a3dd075033f165fc7b3a5991266dd6c6866bae27083b320c4fbd2047182c802e9",
2022-06-10T15:24:01.1043044Z             "70ced2f16c32dcc85e9701d4c05497e32bfe5cc2230b71ba5ee5544a2be52c6a46",
2022-06-10T15:24:01.1043443Z             "64bcbdbe35e500af5f45689760422d75f0d98a95fe41ebc82b2237189845408007",
2022-06-10T15:24:01.1043907Z             "097ad5b5b680c1b3318ffd82c3c467375206d50251460b3aaae5f5d141734e6880",
2022-06-10T15:24:01.1044287Z             "7b9131be8b227a84692cf206711afc4b89726867b2ac58466784db1f1f44d74335"
2022-06-10T15:24:01.1044567Z         ],
2022-06-10T15:24:01.1044763Z         "unseal_shares": 5,
2022-06-10T15:24:01.1044998Z         "unseal_threshold": 3
2022-06-10T15:24:01.1045394Z     }
2022-06-10T15:24:01.1045560Z }

Now:

2022-08-25T13:15:34.9263059Z TASK [vault : Print vault keys] ************************************************
2022-08-25T13:15:34.9582656Z ok: [localhost] => {
2022-08-25T13:15:34.9582950Z     "vault_keys": {
2022-08-25T13:15:34.9583179Z         "changed": true,
2022-08-25T13:15:34.9583472Z         "failed": false,
2022-08-25T13:15:34.9583725Z         "keys": [
2022-08-25T13:15:34.9584049Z             "2a3ad851f8a56b94d8eae58ec51c4f21b8061916da0974afbc205fe83911663b08",
2022-08-25T13:15:34.9584880Z             "7aad96a0ac09caf3e73ff14acccd9d8c4177326a800ade8495ddd6c455d4a859b1",
2022-08-25T13:15:34.9585348Z             "66a0c853825995d92adbb25421fde4d8e39694d1b0013d88747b8eab6c4934a674",
2022-08-25T13:15:34.9585750Z             "0db0401e0d9135e2157ea26576f308ad7ff9c489f6989a662297599f4fa5452a2e",
2022-08-25T13:15:34.9586142Z             "0800e29fbedb403dde923056ba76134c7352ea53119ebbd462b79828f876889758"
2022-08-25T13:15:34.9586443Z         ],
2022-08-25T13:15:34.9586662Z         "keys_base64": [
2022-08-25T13:15:34.9586996Z             "KjrYUfila5TY6uWOxRxPIbgGGRbaCXSvvCBf6DkRZjsI",
2022-08-25T13:15:34.9587409Z             "eq2WoKwJyvPnP/FKzM2djEF3MmqACt6Eld3WxFXUqFmx",
2022-08-25T13:15:34.9588719Z             "ZqDIU4JZldkq27JUIf3k2OOWlNGwAT2IdHuOq2xJNKZ0",
2022-08-25T13:15:34.9589115Z             "DbBAHg2RNeIVfqJldvMIrX/5xIn2mJpmIpdZn0+lRSou",
2022-08-25T13:15:34.9589518Z             "CADin77bQD3ekjBWunYTTHNS6lMRnrvUYreYKPh2iJdY"
2022-08-25T13:15:34.9589823Z         ],
2022-08-25T13:15:34.9590279Z         "rc": 0,
2022-08-25T13:15:34.9591916Z         "root_token": "hvs.0hH5aSVtNLPv1GtJVbJugDgA"
2022-08-25T13:15:34.9592196Z     }
2022-08-25T13:15:34.9592372Z }

Variable names in the dict changed, but look similar (keys == unseal_keys_hex, keys_base64 == unseal_keys_b64), shouldn't be a big of a problem for existing deployments (given they are already initialised).

Reply:
We could make it compatible. Unsure if it's worth it if the information is still there. Are there playbooks out there using this variable?

Reply:
I have never written such, so I believe there are none.